libmscore/system.cpp

```diff
@@ -642,11 +642,14 @@ void System::layout2()
             // if end staff not visible, try prev staff
             while (staffIdx1 <= staffIdx2 && !_staves[staffIdx2]->show())
                   --staffIdx2;
+            // it's user's choice to decide spanning to how many staves qualifies a bracket for being shown
+            // the decision is a score style "bracketMinStaves", referenced as BMS below
             // the bracket will be shown IF:
-            // it spans at least 2 visible staves (staffIdx1 < staffIdx2) OR
-            // it spans just one visible staff (staffIdx1 == staffIdx2) but it is required to do so
+            // it spans at least BMS visible staves OR
+            // it spans just less than BMS visible staff but it is required to do so
             // (the second case happens at least when the bracket is initially dropped)
-            bool notHidden = (staffIdx1 < staffIdx2) || (b->span() == 1 && staffIdx1 == staffIdx2);
+            int bms = score()->styleI(Sid::bracketMinStaves);
+            bool notHidden = (staffIdx1 < staffIdx2 + (2 - bms)) || (staffIdx1 == staffIdx2 + 1 - b->span());
```

Comment:
This isn't the correct logic - not as I proposed it, anyhow.  It needs to be about the total number of instruments (or possibly staves) in the score, not the number of staves - visible or otherwise - spanned by this bracket.  This won't produce the results I described at all, as far as I can tell.

Reply:
Updated as you proposed.